File: array/0001_two_sum/0001_two_sum.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) []int {
	index := make(map[int]int, len(nums))

	for i, b := range nums {
		if j, ok := index[target-b]; ok {
			return []int{j, i}
		}

		index[b] = i
	}

	return nil
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Println(twoSum(nums, target))
}

File: array/0001_two_sum/0001_two_sum.py
# -*- coding: utf-8 -*-
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for idx, num in enumerate(nums):
            if (target - num) in d:
                return [d[target-num], idx]
            d[num] = idx


nums = [2,7,11,15]
target = 9
print(Solution().twoSum(nums, target))